test/core/xds/xds_cluster_resource_type_test.cc

```diff
@@ -1409,6 +1411,35 @@ TEST_F(HostOverrideStatusTest, CanExplicitlySetToEmpty) {
   EXPECT_EQ(resource.override_host_statuses.ToString(), "{}");
 }
 
+using TelemetryLabelTest = XdsClusterTest;
+
+TEST_F(TelemetryLabelTest, ValidConfig) {
+  Cluster cluster;
+  cluster.set_type(cluster.EDS);
+  cluster.mutable_eds_cluster_config()->mutable_eds_config()->mutable_self();
+  ::google::protobuf::Struct telemetry_labels;
```

Comment:
I think this can be written a bit more idiomatically, and with less unnecessary copying, using something like this:

```
auto& filter_map = *cluster.mutable_metadata()->mutable_filter_metadata();
auto& label_map = *filter_map["com.google.csm.telemetry_labels"]->mutable_fields();
*label_map["service_name"].mutable_string_value() = service_name;
*label_map["service_namespace"].mutable_string_value() = service_namespace;
```